### ISSUE TITLE
[SCREENREADER]: Subway list steps SHOULD announce the step number

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/vets-design-system-documentation#247]
@jenstrickland commented on [Mon Jun 08 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936)

# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

```diff
! Team affected: VSA BAM2; Project: MDOT MVP
```

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

The introduction page for various VA applications use a "subway list" pattern. This text generally of the numbers of the subway list **should** read aloud in a screen reader. On this introduction page, the numbers are not read aloud in JAWS+IE11.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

- [ ] As a screen reader user, I want to know which step of the process I am on, on par with a sighted user's experience.
- [ ] Create and/or modify unit tests for this component to include an aXe scan

## Environment

* Operating System: Windows
* Browser: IE11, Firefox
* Screenreading device: JAWS, NVDA
* Server destination: staging

> Note: VoiceOver+Safari reads the individual steps. I tested another page with the subway steps, Apply for health care benefits, and heard the numbers read by NVDA, but not on this hearing aid batteries intro.

## Steps to Recreate

1. Enter https://staging.va.gov/hearing-aid-batteries-and-accessories/introduction in browser
2. Start screenreading device listed in Environment
3. Begin to read the page
4. Verify that the numbered subway steps are not read aloud, and there is no communication of the progress through the steps for screen reader users.

## Possible Fixes (optional)

See some of the other pages that use the subway list, and use the same code.

## WCAG or Vendor Guidance (optional)

* [wuhcag - consistent identification](https://www.wuhcag.com/consistent-identification/)

* [W3C WCAG - Understanding Success Criterion 3.2.4: Consistent Identification](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html)

---

@mr0sari0 commented on [Tue Jun 09 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-641418162)

I suspect this to be a platform related issue or a Firefox related issue with CSS content that's added before or after the element. I tested other forms on Firefox with Pericles and it skips over the number of the step. In the screenshot below, you can see that the aria-names of the steps include the number.

<img width="892" alt="Screen Shot 2020-06-09 at 12 25 59 PM" src="https://user-images.githubusercontent.com/12755283/84174698-ee45a700-aa4c-11ea-816b-19062dc7cbd0.png">

---

@jenstrickland commented on [Tue Jun 09 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-641447724)

@1Copenut — Have you heard of this bug previously? Is this known behavior for the subway list numbers?

I tested `Apply for health care benefits` with Firefox+NVDA and heard each of the numbers read out. However, for this work, the numbers are only read aloud in VoiceOver+Safari.

---

@mr0sari0 commented on [Wed Jun 10 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-642186633)

@jenstrickland I just retested this with VoiceOver and Safari and I didn't hear the numbers read aloud for this form and the health care application.

---

@jenstrickland commented on [Wed Jun 10 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-642238574)

@mr0sari0 Here's a [link to a recording of the screen in VoiceOver+Safari](https://dsva.slack.com/archives/CPE4AJ6Q0/p1591820693054200), in the BAM2 Slack channel.

---

@1Copenut commented on [Wed Jun 10 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-642250441)

@jenstrickland && @mr0sari0 We're probably going to have to add a `role="list"` to the [subway map markup](https://design.va.gov/components/process-list) based on the design.va.gov component. It's using css to defeat the list-styling, which is likely the culprit for Safari + Voiceover.

https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html has a deep dive into this issue.

```css
.process {
    list-style: none;
    padding: 1em 0;
    position: relative;
}
```

---

@mr0sari0 commented on [Wed Jun 10 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-642258053)

> @mr0sari0 Here's a [link to a recording of the screen in VoiceOver+Safari](https://dsva.slack.com/archives/CPE4AJ6Q0/p1591820693054200), in the BAM2 Slack channel.

Thanks for the link @jenstrickland. Is it safe to close this issue if it's working?

---

@jenstrickland commented on [Wed Jun 10 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-642267151)

@1Copenut — The list is reading aloud, correctly, in Voiceover+Safari. 

JAWS+IE11 and Firefox+NVDA are not announcing the list steps in this product's implementation, but other products do. 

---

@mr0sari0 commented on [Wed Jun 10 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-642284798)

> @jenstrickland && @mr0sari0 We're probably going to have to add a `role="list"` to the [subway map markup](https://design.va.gov/components/process-list) based on the design.va.gov component. It's using css to defeat the list-styling, which is likely the culprit for Safari + Voiceover.
> 
> https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html has a deep dive into this issue.
> 
> ```css
> .process {
>     list-style: none;
>     padding: 1em 0;
>     position: relative;
> }
> ```

I just added `role='list'` to my `ol` and still couldn't get the numbers to be read.

---

@1Copenut commented on [Mon Jun 15 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-644310359)

@mr0sari0 In this case, you might have to create code in isolation on a Codepen or other scratch pad, then introduce the CSS rules slowly to see which one breaks the behavior. I'd have thought adding `role="list"` to the OL would have solved it, but I guess that's a UL trick only.

---

@jenstrickland commented on [Mon Jun 15 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-644333246)

@1Copenut  — @mr0sari0 and I hopped on a call and VoiceOver+Safari is reading the numbers. 

Mahariel, let me know if you added the `role="list"` and I can test in JAWS & NVDA.

---

@mr0sari0 commented on [Wed Jun 17 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-645481202)

@1Copenut and @jenstrickland, I am actually getting an error for adding `role="list"`. Apparently, OLs have an implicit role of list.

---

@jenstrickland commented on [Mon Jun 22 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-645527739)

Since the list semantics are removed by the CSS `list-style:none` that would remove it, so the error might have to be accepted. 

I inspected your code in comparison with https://www.va.gov/health-care/apply/application/introduction and it looks the same. 

I just tested both urls with JAWS+IE11, and it looks like it's a platform level issue, so I am going to close this issue for your team — as long as the `role="list"` is not added. Would you please confirm that you are not adding it in a PR?

![Screen Shot 2020-06-17 at 1 49 46 PM](https://user-images.githubusercontent.com/57469/84931917-7c4f0c80-b0a1-11ea-804f-0c0ef75be5f8.png)

---

@mr0sari0 commented on [Fri Jun 19 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-646856612)

Hi @jenstrickland, thanks for looking deeper into this. I have not added the `role="list"` to the PR.

---

@jenstrickland commented on [Mon Jun 22 2020](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9936#issuecomment-647624428)

This is a platform level issue, a JAWS+IE11 bug, so I will assign to platform to take a look.

**Comment by jenstrickland** [comment]
Once this issue is resolved, please notify the VSA BAM2 team so they may audit their work to ensure the issue is resolved. Thanks!

**Comment by ddzz** [comment]
This component has not been moved to `component-library` yet and due to this it cannot be modified easily to make these changes.
